src/test/java/seedu/duke/ViewFunctionTest.java

```diff
@@ -68,84 +73,699 @@ void view_allFieldsValid_success() {
+                    + System.lineSeparator()
+                    + "Here is the list of all recurring entries, where some were added to the above list:"
+                    + System.lineSeparator()
+                    + "Income  |  COMMISSION   | 2021-06-09 |    OnlyFans    | $300.00     | MONTH | 2021-11-06"
+                    + System.lineSeparator()
+                    + "Expense | ENTERTAINMENT | 2012-06-06 |   Maid Cafe    |-$14.60      | YEAR  | 2021-11-06"
+                    + System.lineSeparator();
+            assertEquals(expectedOutput, output.toString());
+        } catch (MintException e) {
+            e.printStackTrace();
+        }
+    }
+
+    @Test
+    void viewByDateAscendingWithRecurringEntries_allFieldsValid_success() {
+
+        NormalFinanceManager normalFinanceManager = new NormalFinanceManager();
+        RecurringFinanceManager recurringFinanceManager = new RecurringFinanceManager();
+        BudgetManager budgetManager = new BudgetManager();
+        NormalListDataManager normalListDataManager = new NormalListDataManager();
+        DataManagerActions dataManagerActions = new DataManagerActions();
+        RecurringListDataManager recurringListDataManager = new RecurringListDataManager();
+        BudgetDataManager budgetDataManager = new BudgetDataManager();
+        Ui ui = new Ui();
+
+        String name = "Samurai Burger";
+        LocalDate date = LocalDate.parse("2021-02-01", dateFormatter);
+        double amount = Double.parseDouble("7.50");
+        ExpenseCategory categoryE = ExpenseCategory.FOOD;
+        Expense expense = new Expense(name, date, amount, categoryE);
+
+        name = "Maid Cafe";
+        date = LocalDate.parse("2012-06-06", dateFormatter);
+        Interval interval = Interval.YEAR;
+        LocalDate endDate = LocalDate.parse("2021-11-06", dateFormatter);
+        amount = Double.parseDouble("14.6");
+        categoryE = ExpenseCategory.ENTERTAINMENT;
+        RecurringExpense expenseR = new RecurringExpense(name, date, amount, categoryE, interval, endDate);
+
+        name = "OnlyFans";
+        date = LocalDate.parse("2021-06-09", dateFormatter);
+        interval = Interval.MONTH;
+        endDate = LocalDate.parse("2021-11-06", dateFormatter);
+        amount = Double.parseDouble("300");
+        IncomeCategory categoryI = IncomeCategory.COMMISSION;
+        RecurringIncome incomeR = new RecurringIncome(name, date, amount, categoryI, interval, endDate);
+
+        name = "Lottery";
+        date = LocalDate.parse("2015-12-15", dateFormatter);
+        amount = Double.parseDouble("250000");
+        categoryI = IncomeCategory.GIFT;
+        Income income = new Income(name, date, amount, categoryI);
+
+        try {
+            normalFinanceManager.addEntry(expense);
+            normalFinanceManager.addEntry(income);
+            recurringFinanceManager.addEntry(expenseR);
+            recurringFinanceManager.addEntry(incomeR);
```

Comment:
Since you are using the same example entries, perhaps you can have a one-time setup function that creates and add all these before all the tests start?